libevm/VMCalls.cpp

```diff
@@ -153,11 +153,20 @@ void VM::caseCall()
 	bytesRef output;
 	if (caseCallSetup(callParams.get(), output))
 	{
-		std::pair<bool, owning_bytes_ref> callResult = m_ext->call(*callParams);
-		callResult.second.copyTo(output);
-		m_returnData.assign(callResult.second.begin(), callResult.second.end());
-
-		m_SPP[0] = callResult.first ? 1 : 0;
+		bool success = false;
+		owning_bytes_ref outputRef;
+		std::tie(success, outputRef) = m_ext->call(*callParams);
+		outputRef.copyTo(output);
```

Comment:
Magic. The `output` is actually a memory reference initialized in `caseCallSetup()`. This is exactly CALL output buffer, might be of length 0 depending on CALL arguments. Here we copy the data there according to what the CALL requested.